File: SuperiorPackGroup/BLL/BOLReportBLL.vb
Imports SuperiorPackGroup.SPGReportsTableAdapters

Public Class BOLReportBLL

    Private m_BOLReportTableAdapter As BOLReportTableAdapter = Nothing

    Public ReadOnly Property Adapter() As BOLReportTableAdapter
        Get
            If (Me.m_BOLReportTableAdapter Is Nothing) Then
                Me.m_BOLReportTableAdapter = New BOLReportTableAdapter
                Me.m_BOLReportTableAdapter.Connection.ConnectionString = My.Settings.UserConnectionString
            End If
            Return Me.m_BOLReportTableAdapter
        End Get
    End Property

    Public Function Getbolreport(ByVal ShippingID As Integer) As SPGReports.BOLReportDataTable

        Try
            Return Adapter.GetBOL(ShippingID)
        Catch ex As Exception
            MessageBox.Show(String.Format("There was an error while retrieving the information to print{0}{1}", vbCrLf, ex.Message))
            Return Nothing
        End Try

    End Function

End Class

File: SuperiorPackGroup/BLL/BOLReportBLL.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using SuperiorPackGroup.SPGReportsTableAdapters;

namespace SuperiorPackGroup
{
	public class BOLReportBLL
	{

		private BOLReportTableAdapter m_BOLReportTableAdapter = null;

		public BOLReportTableAdapter Adapter
		{
			get
			{
				if (m_BOLReportTableAdapter == null)
				{
					m_BOLReportTableAdapter = new BOLReportTableAdapter();
					m_BOLReportTableAdapter.Connection.ConnectionString = Properties.Settings.Default.UserConnectionString;
				}
				return m_BOLReportTableAdapter;
			}
		}

		public SPGReports.BOLReportDataTable Getbolreport(int ShippingID)
		{

			try
			{
				return Adapter.GetBOL(ShippingID);
			}
			catch (Exception ex)
			{
				MessageBox.Show(string.Format("There was an error while retrieving the information to print{0}{1}", Environment.NewLine, ex.Message));
				return null;
			}

		}

	}

}